strawberryfields/configuration.py

```diff
@@ -218,6 +220,106 @@ def parse_environment_variable(key, value):
+    r"""Configure Strawberry Fields for access to the Xanadu Cloud platform by
+    saving your account credentials.
+
+    The configuration file can be created in the following locations:
+     
+     - A global user configuration directory ("user_config")
+     - The current working directory ("local")
+     
+     This global user configuration directory differs depending on the operating system:
+     
+     * On Linux: ``~/.config/strawberryfields``
+     * On Windows: ``~C:\Users\USERNAME\AppData\Local\Xanadu\strawberryfields``
+     * On MacOS: ``~/Library/Application Support/strawberryfields``
+     
+     By default, Strawberry Fields will load the configuration and account credentials from the global
+     user configuration directory, no matter the working directory. However, if there exists a configuration
+     file in the *local* working directory, this takes precedence. The ``"local"`` option is therefore useful
+     for maintaining per-project configuration settings.
+
+    **Examples:**
+
+    In these examples ``"MyToken"`` contains the user-specific authentication
+    token.
+
+    The access to the Xanadu Cloud can be configured simply by running.
+
+    >>> import strawberryfields as sf
+    >>> sf.store_account("MyToken")
+
+    This creates the following ``"config.toml"`` file:
+
+    .. code-block:: toml
+
+        [api]
+        authentication_token = "MyToken"
+        hostname = "localhost"
+        use_ssl = true
+        port = 443
+
+    You can also create the configuration file locally (in the **current
+    working directory**) the following way:
+
+    >>> import strawberryfields as sf
+    >>> sf.store_account("MyToken", location="local")
+
+    Each of the configuration options (check out the
+    :doc:`/introduction/configuration` page for a list of options) can be
+    passed as further keyword arguments as well:
+
+    >>> import strawberryfields as sf
+    >>> sf.store_account("MyToken", location="local", hostname="MyHost", use_ssl=False, port=123)
+
+    This creates the following ``"config.toml"`` file in the **current working directory**:
+
+    .. code-block:: toml
+
+        [api]
+        authentication_token = "MyToken"
+        hostname = "MyHost"
+        use_ssl = false
+        port = 123
+
+    Args:
+        authentication_token (str): API token for authentication to the Xanadu Cloud platform.
+            This is required for submitting remote jobs using :class:`~.StarshipEngine`.
+
+    Kwargs:
+        location (str): determines where the configuration file should be saved
+        filename (str): the name of the configuration file to look for
+
+    Additional configuration options are detailed in :doc:`/introduction/configuration` and can be passed
+    as keyword arguments.
+    """
+    if location == "user_config":
+        directory = user_config_dir("strawberryfields", "Xanadu")
+
+        # Create target Directory if it doesn't exist
+        os.makedirs(directory, exist_ok=True)
```

Comment:
I've added this since an error would be otherwise raised if the configuration directory did not exist.

This solution follows the suggestions from here:
https://stackoverflow.com/questions/273192/how-can-i-safely-create-a-nested-directory